code/util/BaseInteractable.cs

```diff
@@ -0,0 +1,82 @@
+﻿namespace Sandbox.util;
+
+public class CanTriggerResults
+{
+    public CanTriggerResults(bool hit, float dist, BaseInteractable interactable, Vector3 hitpos)
+    {
+        Hit = hit;
+        Distance = dist;
+        Interactable = interactable;
+        HitPosition= hitpos;
+    }
+
+    public bool Hit;
+    public float Distance;
+    public BaseInteractable Interactable;
+    public Vector3 HitPosition;
+}
+
+public partial class BaseInteractable : BaseNetworkable
+{
+    [Net]
+    public Entity Parent { get; set; }
+    [Net]
+    public Vector3 Mins { get; set; }
+    [Net]
+    public Vector3 Maxs { get; set; }
+    public float MaxDistance { get; set; } = 60f;
+    public CanTriggerResults LastTriggerResults {get; set; }
+
+    public BaseInteractable()
+    {
+    }
+
+    public virtual void Trigger(Cinema.Player ply = null)
+    {
+    }
+
+    /// <summary>
+    /// This one will check if you can trigger.
+    /// </summary>
+    /// <param name="ray"></param>
+    /// <returns></returns>
+    public CanTriggerResults CanTrigger(Ray ray)
```

Comment:
This is a very confusing function.

Without looking at the code much, I assume "Can Trigger takes a ray and will return whether the ray will trigger the interactable."

I would assume the result is a boolean.

But instead it's a `CanTriggerResults` which is a struct of a `bool` and other data which describes a trace which was performed to check if a ray will trigger the interactable.

Maybe you could call this `TestRay` or `CanRayTrigger` and you could just call the return struct `TriggerTraceResult` ?